docs(configuration): add explanation of new path definition which includes first/last callback execution (#172)
tier4/CARET_doc: docs/configuration/inter_node_data_path.md

```

An exception is raised for pairs that cannot be combined.

### Considering first/last callback

In CARET, the path is defined as `[node_name]-[topic_name]-... -[topic_name]-[node_name]` (for more information in [Path](../design/event_and_latency_definitions/path.md#Path)).
In the default configuration, the path analysis does not include the callback processing times at both ends of the above paths. The following two processing times are not included by default in path analysis.

> - `callback_start` to `rclcpp_publish` or `rclcpp_intra_publish` in the first node.

Reviewer: ```suggestion
- Processing time in the first node; from `callback_start` to `rclcpp_publish` or `rclcpp_intra_publish`
```

> - `callback_start` to `callback_end` in the last node.

Reviewer: ```suggestion
- Processing time in the last node; from `callback_start` to `callback_end`
```

We provide options for evaluating the callback times at the end of these paths.

- `Path.include_first_callback`
  - Default: `False`

>   - If `True`, the time from callback_start to publish is included in the path analysis. The trace data used for the above callback_start adopts the data closest to the time of publish among those with the same tid as publish. Note that the first callback time added here may contain more than one callback object.

Reviewer: ```suggestion
  - If `True`, the processing time in the first node is included in the path analysis. The callback_start in the first node is that of a trace data whose timestamp is the closest to publish and thread id (tid) is the same as publish. Note that a callback object of the callback_start in the first node can vary if the node has several callbacks and any of them runs before publish.
```

- `Path.include_last_callback`
  - Default: `False`

>   - If `True`, the time from callback_start to callback_end is included in the path analysis.

Reviewer: ```suggestion
  - If `True`, the processing time in the last node is included in the path analysis.
```
